### ISSUE TITLE
Docker provider fails to connect via SSH or non-TLS TCP on WSL (Windows)

### ISSUE THREAD

**Issue by badgerwithagun** [terraform-providers/terraform-provider-docker#210]
### Terraform Version
Terraform v0.12.10
+ provider.docker v2.4.0

### Affected Resource(s)
Provider

### Terraform Configuration Files
```
provider "docker" {
  host = "ssh://root@myhost.example.com:1234"
}
```
(note that the address and port are faked in this example).

### Steps to Reproduce
1. `terraform init`
1. `terraform apply`

### Expected Behavior
The following works:
```
docker -H ssh://root@myhost.example.com:1234 run -ti alpine echo "Hello"
```
So Docker is connecting just fine via SSH (no public key issues or anything). Therefore I'd expect the above configuration to work.

### Actual Behavior
```
Error: Error pinging Docker server: error during connect: Get https://root%40myhost.example.com:1234/_ping: dial tcp: lookup root@myhost.example.com: no such host
```
It's picking up the host name correctly, but the `https` is suspicious.

### Debug Output
https://gist.github.com/badgerwithagun/5020bdf6daab139716b2125f927413c1
(Note - search & replaced real hostname, paths etc to match the example - please just take it as assumed that I've checked that stuff!)

### Panic Output
None

### Important Factoids
* Client is Ubuntu 18.04.3 LTS on Windows Subsystem for Linux
* This also occurs if I remove the `host` in the provider definition and instead set `DOCKER_HOST` before calling `terraform apply`.  Terraform picks up the host name correctly from `DOCKER_HOST`, but still produces the same error message implying it's not using SSH.

### References
None

**Comment by badgerwithagun** [comment]
If anyone comes across this, I solved it.

Like most people using WSL _and_ Docker, I had set `docker` on WSL to use the Docker For WIndows daemon by default, which uses TLS.  I had the following in my `~/.bash_profile` (but I guess you might have it in `.bashrc` or even `/etc/profile`):
```
export DOCKER_TLS_VERIFY=1
export DOCKER_HOST=tcp://1.2.3.4:2376
export DOCKER_CERT_PATH=/mnt/c/Users/myuser/.docker/machine/machines/default
export DOCKER_MACHINE_NAME=default
```
I guess `DOCKER_TLS_VERIFY` is the main issue.  I did the following:
```
export DOCKER_TLS_VERIFY=
export DOCKER_HOST=
terraform apply
```
And it worked.  A permanent solution is just to uncomment the relevant lines in the profile script, but that means docker stops working locally.
